NeoMathEngine/src/CPU/x86/AvxDll.h

```diff
@@ -0,0 +1,69 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <array>
+#include <NeoMathEngine/NeoMathEngine.h>
+#include <MathEngineDnnConv.h>
+#include <MathEngineDll.h>
+
+namespace NeoML {
+
+// Class support dynamic library which is compiled with AVX for performance increasing.
+class CAvxDll : public CDll {
+public:
+	CAvxDll( const CAvxDll& ) = delete;
+	CAvxDll& operator=( const CAvxDll& ) = delete;
+	CAvxDll( CAvxDll&& ) = delete;
+	CAvxDll& operator=( CAvxDll&& ) = delete;
+
+	static CAvxDll& GetInstance();
```

Comment:
The library object has a static storage duration. Perhaps it is better to unload the library in the 'MathEngine' destructor using CDllLoader?

Reply:
I don't think that we should unload "*avx.so". Because it's not a third-party library! It's actually a part of MathEngine.